src/Util/MIMETypeUtil.m

```diff
@@ -277,7 +277,65 @@ + (BOOL)isAudio:(NSString *)contentType {
 
 + (NSString *)filePathForAttachment:(NSString *)uniqueId
                          ofMIMEType:(NSString *)contentType
-                           inFolder:(NSString *)folder {
+                           filename:(nullable NSString *)filename
+                           inFolder:(NSString *)folder
+{
+    NSString *kDefaultFileExtension = @"bin";
```

Comment:
We repeatedly call this method to determine what the filepath for a given attachment is. Instead, we should call it once, and store the file path on the attachment model.  That would make us robust to changes to this method's behavior.